src/swap/defi/thorchain.ts

```diff
@@ -961,3 +1027,57 @@ const convertChainAmountToAsset = (
   const out = div(mul(amount, sourceRate), destRate, DIVIDE_PRECISION)
   return out
 }
+
+export const getVolatilitySpread = (
+  log: EdgeLog,
+  {
+    fromPluginId,
+    fromTokenId,
+    fromCurrencyCode,
+    toPluginId,
+    toTokenId,
+    toCurrencyCode,
+    likeKindVolatilitySpread,
+    volatilitySpread,
+    perAssetSpread
+  }: {
+    fromPluginId: string
+    fromTokenId?: string
+    fromCurrencyCode: string
+    toPluginId: string
+    toTokenId?: string
+    toCurrencyCode: string
+    likeKindVolatilitySpread: number
+    volatilitySpread: number
+    perAssetSpread: AssetSpread[]
+  }
+): string => {
+  let volatilitySpreadFinal: number | undefined
+
+  for (const spread of perAssetSpread) {
+    const {
+      sourcePluginId,
+      sourceTokenId,
+      destPluginId,
+      destTokenId,
+      volatilitySpread
+    } = spread
+    if (sourcePluginId != null && sourcePluginId !== fromPluginId) continue
+    if (sourceTokenId != null && sourceTokenId !== fromTokenId) continue
+    if (destPluginId != null && destPluginId !== toPluginId) continue
+    if (destTokenId != null && destTokenId !== toTokenId) continue
+    volatilitySpreadFinal = volatilitySpread
+    break
+  }
+
+  if (volatilitySpreadFinal == null) {
+    const likeKind = isLikeKind(fromCurrencyCode, toCurrencyCode)
+
+    volatilitySpreadFinal = likeKind
+      ? likeKindVolatilitySpread
+      : volatilitySpread
+  }
+
+  log.warn(`getVolatilitySpread: ${volatilitySpreadFinal.toString()}`)
```

Comment:
This should be info, not a warning. Nothing has gone wrong, and verbose logging can activate it if needed. It should also be moved out of this routine into the `fetchSwapQuote` body, since unit tests should *never* write to the console.

Yes, we don't enforce this very well, but a test's only job is to pass or fail, not produce output. If you're trying to debug a failing test, and some irrelevant neighbor is being noisy, that's not helpful.

Reply:
Happy to move it out to fetchSwapQuote, but I used log.warn because I want to make sure we get this value in the logs if a swap ends up failing due to spread. This is a debug value that only helps after the fact. If user runs into a failed swap, turning on verbose wont help as by then it's too late.

Reply:
Makes sense. Approved.